core/src/main/java/org/mule/runtime/core/internal/connection/PoolingConnectionHandler.java

```diff
@@ -68,9 +72,11 @@ public void release() {
 
     boolean returnAttempted = false;
     try {
+      LOGGER.trace("Returning connection {} to pool {}", connection.toString(), poolId);
```

Comment:
It's not pretty I know, but the connection instance could be useful for tracking and debugging issues. As you said, because the connection class is created by developers, we can't guarantee nor enforce that it will have a toString() method. But I think that this is better than nothing.